### PR TITLE
AHOYAPPS-446: Community login error handling

### PR DESCRIPTION
**Pull request by Alton09**
## Description

Added detailed error handling for the CommunityLoginActivity.

passcode_expired

passcode_incorrect

## Breakdown

- Added TextInputLayout passcode errors for invalid and expired token errors.
- Validated AuthService response success and error body messages
- Pulled out community specific LoginResult types from internal flavor

## Validation

- Manual validation
- Passed CI pipeline

## Additional Notes

N/A

## Submission Checklist
 - [x] The source has been evaluated for semantic versioning changes and are reflected in ```versionName``` under `app/build.gradle`
 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code

### CODE REVIEW

File: app/lint-baseline.xml [hunk 1]
```diff
@@ -1,6 +1,19 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <issues format="5" by="lint 3.5.0" client="gradle" variant="all" version="3.5.0">
 
+    <issue
```
Comment (aaalaniz) [inline app/lint-baseline.xml]: Is there a better way to manage lint warnings/errors? We have adopted the habit of just updating the baseline each time we have an error which to my understanding is not how the baseline file is intended to be used.

File: app/src/community/java/com/twilio/video/app/ui/login/CommunityLoginActivity.java [hunk 2]
```diff
@@ -65,10 +72,16 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         setContentView(R.layout.development_activity_login);
+
         ButterKnife.bind(this);
         if (authenticator.loggedIn()) startLobbyActivity();
     }
 
+    @Override
```
Comment (aaalaniz) [inline app/src/community/java/com/twilio/video/app/ui/login/CommunityLoginActivity.java]: Can we remove this onResume override?

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -7,6 +7,11 @@
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
     </style>
+
+    <style name="ErrorTextAppearance">
```
Comment (aaalaniz) [inline app/src/main/res/values/styles.xml]: Going back to my comment about managing lint errors. It appears to me that the lint error could be resolved by simply introducing this style under the community variant resources.

File: app/src/testCommunity/java/com/twilio/video/app/ui/login/CommunityLoginActivityTest.kt [hunk 4]
```diff
@@ -105,55 +109,79 @@ class CommunityLoginActivityTest {
...
+        clickLoginButton()
+
+        assertErrorDialogIsDisplayed()
     }
 
-    @Ignore("Will be implemented as part of https://issues.corp.twilio.com/browse/AHOYAPPS-446")
     @Test
     fun `it should enable the login button after all required fields have been entered`() {
-        TODO("not implemented")
+        assertLoginButtonIsDisabled()
```
Comment (aaalaniz) [inline app/src/testCommunity/java/com/twilio/video/app/ui/login/CommunityLoginActivityTest.kt]: Nice 👍

File: app/lint-baseline.xml (hunk 1 above)
Comment (Alton09) [inline app/lint-baseline.xml]: This is a false positive. The style is used by the community login activity layout.

Comment (aaalaniz) [reply app/lint-baseline.xml]: See my comment below. I do not think this is a false positive. The issue states that the resource is unused in some variants which is most certainly true.

Comment (Alton09) [reply app/lint-baseline.xml]: No, I think what you have in place is great! I do my best to fix the lint error before simply adding it to the baseline, but in this case I believe it is a false positive. I'll try out your suggestion below to see if that resolves it.

File: app/src/main/res/values/styles.xml (hunk 3 above)
Comment (Alton09) [reply app/src/main/res/values/styles.xml]: Ah yep that fixed it. Sorry, somehow I totally overlooked the ```includedVariants``` line and thought it was obviously being used in the community variant 😆. Good catch 👍

**Comment by Alton09** [comment]
> Will there be a follow up PR to address an error that results when fetching a token before joining the room. I would expect for the community user to be logged out and navigated back to the login screen in the event that the passcode has expired or is invalid.

Can we cut a follow up ticket for this and I can start working on it right away if needed? This PR is independent of those changes and can get merged in with no issues.

**Pull request merged**